.github/workflows/linux.yml

```diff
@@ -80,6 +81,20 @@ jobs:
         [ ! -f ${PATCH_PREFIX}.patch ] || patch -p1 --no-backup-if-mismatch < ${PATCH_PREFIX}.patch
         [ ! -f ${PATCH_PREFIX}-${{ matrix.build-type }}.patch ] || patch -p1 --no-backup-if-mismatch < ${PATCH_PREFIX}-${{ matrix.build-type }}.patch
 
+    - name: Set up vcpkg archive cache
+      id: setup-vcpkg-cache
+      run: |
+        mkdir --parents ${{ env.VCPKG_INSTALL_CACHE }}
+        sg lxd -c '/snap/bin/lxc profile device add default vcpkg disk source=${{ env.VCPKG_INSTALL_CACHE }} path=/root/parts/multipass/build/vcpkg_installed'
+
+    - name: Set up vcpkg
+      uses: lukka/run-vcpkg@v7.1
+      with:
+        setupOnly: true
+        vcpkgDirectory: '${{ github.workspace }}/3rd-party/vcpkg'
+        appendedCacheKey: ${{ runner.os }}-${{ matrix.build-type }}-${{ hashFiles( '**/vcpkg.json' ) }}
+        additionalCachedPaths: ${{ env.VCPKG_INSTALL_CACHE }}
```

Comment:
Does this make sense? Do we need vcpkg outside of the LXD container?

If not, you can use the cache action directly:
```yaml
    - name: CCache
      uses: actions/cache@v2
      with:
        key: vcpkg-${{ runner.os }}-${{ matrix.build-type }}-${{ hashFiles( '**/vcpkg.json' ) }}
        path: ${{ env.VCPKG_INSTALL_CACHE }}
```

Reply:
As discussed in chat, the `run-vcpkg` action uses the `cache` action internally.

Reply:
More importantly, the `vcpkg` executable gets built in this step, and is shipped along with the source into the snap. Which I dislike a lot ;)

If there's a way to change where vcpkg gets built, in a way that makes it not pollute the source, but rather it getting shipped along with `vcpkg_installed`, that would be nice.

Reply:
Ok, I see your point and will see if there is a way to massage `vcpkg` to not build directly in its submodule directory.  If so, then I may be able to dump the `run-vcpkg` action and just use the `cache action` along with any of our own logic.